dpnp/dpnp_algo/dpnp_algo.pyx

```diff
@@ -98,6 +98,10 @@ cpdef dparray dpnp_array(obj, dtype=None):
     cdef elem_dtype
     cdef dparray_shape_type obj_shape
 
+    # convert scalar to tuple
+    if dpnp.isscalar(obj):
```

Comment:
maybe it is better to use https://github.com/IntelPython/dpnp/blob/dea4c7a2712bbb1c60b58e7b142cd318ef7a6641/dpnp/dpnp_utils/dpnp_algo_utils.pyx#L296?

Reply:
I think that in this place it is better to explicitly show that we convert scalar to tuple tobe able to work with it as with sequence object.
BTW _object_to_tuple(object obj) will not work with all scalars, I made a separate PR for this https://github.com/IntelPython/dpnp/pull/512